From 0b25e4da23d209f96ec0503b5d391cf34e9a67f0 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Thu, 30 Dec 2010 10:16:15 +0800
Subject: [PATCH] ENGR00137472-2 GPMI : remove mil_set_mtd_geometry()

This function is used to re-calculate the parameters for NAND and MTD.
But because we have used the ->init_size() to initialize these parameters.
So this function is useless, just remove it.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c |   52 ------------------------------
 1 files changed, 0 insertions(+), 52 deletions(-)

diff --git a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c
index 1f13b38..66bdb47 100644
--- a/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c
+++ b/drivers/mtd/nand/gpmi-nfc/gpmi-nfc-mil.c
@@ -1080,55 +1080,6 @@ static int mil_set_boot_rom_helper_geometry(struct gpmi_nfc_data  *this)
 }
 
 /**
- * mil_set_mtd_geometry() - Set up the MTD geometry.
- *
- * This function adjusts the owning MTD data structures to match the logical
- * geometry we've chosen.
- *
- * @this:  Per-device data.
- */
-static int mil_set_mtd_geometry(struct gpmi_nfc_data *this)
-{
-	struct physical_geometry  *physical = &this->physical_geometry;
-	struct mil                *mil      = &this->mil;
-	struct nand_ecclayout     *layout   = &mil->oob_layout;
-	struct nand_chip          *nand     = &mil->nand;
-	struct mtd_info           *mtd      = &mil->mtd;
-
-	/* Configure the struct nand_ecclayout. */
-	layout->eccbytes          = 0;
-	layout->oobavail          = physical->page_oob_size_in_bytes;
-	layout->oobfree[0].offset = 0;
-	layout->oobfree[0].length = physical->page_oob_size_in_bytes;
-
-	/* Configure the struct mtd_info. */
-	mtd->size        = nand->numchips * physical->chip_size_in_bytes;
-	mtd->erasesize   = physical->block_size_in_bytes;
-	mtd->writesize   = physical->page_data_size_in_bytes;
-	mtd->ecclayout   = layout;
-	mtd->oobavail    = mtd->ecclayout->oobavail;
-	mtd->oobsize     = mtd->ecclayout->oobavail + mtd->ecclayout->eccbytes;
-	mtd->subpage_sft = 0; /* We don't support sub-page writing. */
-
-	/* Configure the struct nand_chip. */
-	nand->chipsize         = physical->chip_size_in_bytes;
-	nand->page_shift       = ffs(mtd->writesize) - 1;
-	nand->pagemask         = (nand->chipsize >> nand->page_shift) - 1;
-	nand->subpagesize      = mtd->writesize >> mtd->subpage_sft;
-	nand->phys_erase_shift = ffs(mtd->erasesize) - 1;
-	nand->bbt_erase_shift  = nand->phys_erase_shift;
-	nand->oob_poi          = nand->buffers->databuf + mtd->writesize;
-	nand->ecc.layout       = layout;
-	if (nand->chipsize & 0xffffffff)
-		nand->chip_shift = ffs((unsigned) nand->chipsize) - 1;
-	else
-		nand->chip_shift =
-				ffs((unsigned) (nand->chipsize >> 32)) + 32 - 1;
-
-	return 0;
-}
-
-/**
  * mil_set_geometry() - Set up the medium geometry.
  *
  * @this:  Per-device data.
@@ -1154,9 +1105,6 @@ static int mil_set_geometry(struct gpmi_nfc_data  *this)
 	if (mil_set_boot_rom_helper_geometry(this))
 		return -ENXIO;
 
-	if (mil_set_mtd_geometry(this))
-		return -ENXIO;
-
 	/*
 	 * Allocate the page buffer.
 	 *
-- 
1.5.4.4

